singer_sdk/sinks/core.py

```diff
@@ -67,46 +67,36 @@ def validate(self, record: dict[str, t.Any]) -> None:
         """
 
 
-class JSONSchemaValidator(BaseJSONSchemaValidator):
+class FastJSONSchemaValidator(BaseJSONSchemaValidator):
     """Validate records using the ``fastjsonschema`` library."""
 
     def __init__(
         self,
         schema: dict,
         *,
         validate_formats: bool = False,
-        format_checker: jsonschema.FormatChecker | None = None,
+        format_checker: dict[str, t.Callable] | None = None,
     ):
         """Initialize the validator.
 
         Args:
             schema: Schema of the stream to sink.
             validate_formats: Whether JSON string formats (e.g. ``date-time``) should
                 be validated.
-            format_checker: User-defined format checker.
+            format_checker: User-defined format validators.
 
         Raises:
             InvalidJSONSchema: If the schema provided from tap or mapper is invalid.
         """
-        jsonschema_validator = jsonschema.Draft7Validator
-
         super().__init__(schema)
-        if validate_formats:
-            format_checker = format_checker or jsonschema_validator.FORMAT_CHECKER
-        else:
-            format_checker = jsonschema.FormatChecker(formats=())
-
         try:
-            jsonschema_validator.check_schema(schema)
-        except jsonschema.SchemaError as e:
+            self.validator = fastjsonschema.compile(
+                self.schema, use_formats=validate_formats, formats=format_checker or {}
+            )
+        except fastjsonschema.JsonSchemaDefinitionException as e:
             error_message = f"Schema Validation Error: {e}"
```

Comment:
I missed this in the previous PR

```suggestion
            error_message = "Schema Validation Error"
```

that way you don't even need to update the tests.